Base 1.3. Верстка главной страницы

tUnknownLegend opened the pull request.

Liza1040 committed:
Добавила страницу регистрации

tUnknownLegend committed:
исправления для .form__input

+border-left-with. Так же цвета вынесены в константы

tUnknownLegend committed:
сделана основа для хедера

Нужно привести css в адекватный вид, расположить кнопки равнее, перекрасить svg и добавить лого в hi-res

Liza1040 committed:
Поправила header: выровнила элементы, поправила стили

thelvv committed:
Добавлена страница логина и частично главная, шаблоны распределены по файлам

tUnknownLegend committed:
главная страница готова

так же добавлен цвет фона в логине и регистрации в боковых дивах

tUnknownLegend committed:
добавлены теги header и footer, длинна страницы теперь своя у каждой страницы

tUnknownLegend committed:
исправление ширины для поля поиска

tUnknownLegend committed:
главная страница

Liza1040 committed:
Поправила стили с учетом базовых стилей кнопки, цветов и шрифтов

tUnknownLegend committed:
унификация css классов

DPeshkoff commented on index.css:
```diff
@@ -0,0 +1,211 @@
+    --apple-shadow: drop-shadow(0 0 20px var(--shadow-default));
+    --shadow-dim: #000000FF;
+    --apple-dim-shadow: drop-shadow(0 4 20px  var(--shadow-dim));
+}*/
```
Закомментированных стилей быть не должно

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+
+<body>
+<div class="page">
+    <div></div>
```
Зачем нужен этот пустой дивак?

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+<div class="page">
+    <div></div>
+    <header>
+        <div class="header">
```
Зачем в header еще один div? Класс можно и к header применить с:

DPeshkoff commented on main.css:
```diff
@@ -0,0 +1,247 @@
+.header__logotip {
+    margin-left: 20px;
+    width: 101px;
+    height: 38px;
```
Интересные цифры. Почему так? 101 вообще простое число, ничему не кратно (а в дизайне лучше использовать [кратные значения ](https://habr.com/ru/company/productivity_inside/blog/319700/)

DPeshkoff commented on main.css:
```diff
@@ -0,0 +1,247 @@
+    border-radius: 10px;
+}
+
+.header__form-search {
```
Неплохо бы еще шрифт сюда - сейчас в инпуте, кажется, дефолтный

DPeshkoff commented on main.css:
```diff
@@ -0,0 +1,247 @@
+}
+
+.header__catalog {
+    align-self: center;
```
А зачем этот стиль? Есть свойство [align-items](https://developer.mozilla.org/en-US/docs/Web/CSS/align-items), руками можно не ставить =)

DPeshkoff commented on main.css:
```diff
@@ -0,0 +1,247 @@
+}
+
+.header__form-search:focus {
+    outline: none !important;
```
1 Почему на фокусе нет аутлайна? 

2 Использовать !important не очень хорошо (кроме тех случаев, когда это прям железно необходимо). Можно почитать [тут](https://tproger.ru/translations/stop-using-important/)

DPeshkoff commented on main.css:
```diff
@@ -0,0 +1,247 @@
+    width: 65px;
+    height: 53px;
+    margin-left: 22px;
+    padding: 5px;
```
Тоже не кратные значения...

DPeshkoff commented on main.css:
```diff
@@ -0,0 +1,247 @@
+}
+
+.header__favourites {
+    width: 91px;
```
... еще и разные для разных блоков. Почему так?

DPeshkoff commented on main.css:
```diff
@@ -0,0 +1,247 @@
+    border-radius: 10px;
+    border: none;
+    font-family: 'IBM Plex Sans Condensed', sans-serif;
+    font-style: normal;
```
По умолчанию он и так normal

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+        </div>
+
+    </header>
+    <div></div>
```
Кто такой?

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+            <div class="header__right">
+                <div class="header__order">
+                    <a href="">
+                        <img src="./img/Orders.svg" alt="LOGO">
```
очень странно, что некоторые svg у вас с заглавной буквы, а другие - со строчных

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+                </div>
+                <div class="header__catalog">
+                    <a href="" class="button-default header__button-catalog paint-purple">
+                        <img type="image" src="./img/Catalog.svg" class="header__catalog-img" alt="CAT">
```
[Что писать в alt](https://habr.com/ru/post/341810/)

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+                <div class="header__search-line">
+                    <input type="search" class="header__form-search" name="search-line"
+                           placeholder="Что Вас интересует?" maxlength="75">
+                    <a href="" class="header__button-search paint-purple">
```
Не особо понимаю смысл в отдельном классе paint-purple. Разъясните, почему так, а не навешивать в button-search тот же background-color на переменную с цветом?

tUnknownLegend replied:
paint-purple нужен, чтобы сайдбар красить. На кнопку навешиваем, чтобы зря не копипастить в css

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+                        <img src="./img/Favourites.svg" alt="LOGO">
+                    </a>
+                </div>
+                <div class="header__card">
```
```suggestion
                <div class="header__cart">
```

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+
+    </header>
+    <div></div>
+    <div class="paint-background"></div>
```
А это кто такой?)

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+    </header>
+    <div></div>
+    <div class="paint-background"></div>
+    <div class="content">
```
Ну то есть [main](http://htmlbook.ru/html/main)?

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+        <div class="content__catalog">
+            <a class="item-top-cards text-normal-default" href="">
+                <img src="./img/Smartphone.png" alt="PHONES">
+                <p>Телефоны</p>
```
Стили к тексту не соответствуют макетам - не то со шрифтами

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+
+            <a class="item-top-cards text-normal-default" href="">
+                <img src="./img/Headphones.png" alt="PHONES">
+                <span>Наушники</span>
```
Что-то с alt

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+        </div>
+
+        <div class="content__sales-title text-normal-huge">
+            Осталось скидки
```
Формулировка довольно странно звучит. Может лучше "Акция"?

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+
+        <div class="content__sales">
+
+            <div class="">
```
У вас кто-то украл класс

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+                        167990₽
+                    </span>
+                        <span class="text-normal-default item-card__title">
+                       title for this fucking card Apple iPhone 13 64Gb
```
На РК такие плейсхолдеры не оставляйте только)

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+                        167990₽
+                    </span>
+                        <span class="text-normal-default item-card__text-strikethrough">
+                        167990₽
```
В макетах есть отступ перед знаком рубля, можно попробовать хотя бы так 
```suggestion
                        167990 ₽
```

tUnknownLegend replied:
убрал в макете отступ

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+                </a>
+
+                <div class="item-card__bottom">
+                    <button type="button" class="button-default button_add-to-card" value="В корзину">В корзину
```
А для чего тут value?

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+        </div>
+
+        <div class="content__popular-title text-normal-huge">
+            Популярное
```
Мне кажется, или тоже визуально не соответствует макетам?

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+    <div class="paint-background"></div>
+    <footer>
+        <div class="placeholder paint-background">
+
```
1 Давайте что-нибудь сюда напишем)

2 А это нормально, что оно выглядит вот так (внимание на область слева и справа)? Так и задумано? В макетах главной, кажется, это не отражено

![изображение](https://user-images.githubusercontent.com/31134795/193413247-fa2aedeb-909d-43b7-bcf5-474e281b17c7.png)

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,530 @@
+</div>
+</body>
+
+</html>
```
Сюда бы trailing newline \n

DPeshkoff commented on main.css:
```diff
@@ -0,0 +1,247 @@
+    --hover_purple: #3F47D0;
+    --pure_white: #FFFFFF;
+    --background-white: #EEEEEE;
+    --placeholder: rgba(55, 55, 55, 0.7);
+    --jet-black: #333333;
```
Давайте для удобства один общий стиль - либо RGBA, либо HEX

лайк за переменные

DPeshkoff commented on main.css:
```diff
@@ -0,0 +1,247 @@
+    border-color: rgba(0, 0, 0, 0);
+    width: 550px;
+    height: 40px;
+    background: #F8F8F8;
```
Этот цвет есть здесь и в `.form__input` 

Если цвета переиспользуются - лучше тоже выносить в переменные

DPeshkoff commented on main.css:
```diff
@@ -0,0 +1,247 @@
+}
+
+.link:hover {
+    color: var(--hover_purple)
```
не пренебрегайте ;

DPeshkoff commented on index.css:
```diff
@@ -0,0 +1,211 @@
+    margin-right: 10px;
+    background: var(--default_purple);
+    color: var(--background-white);
+}
```
\n

DPeshkoff commented on index.css:
```diff
@@ -0,0 +1,211 @@
+}
+
+.item-card {
+    text-decoration: none;
+    width: 165px;
+    height: 312px;
+    padding: 15px 18px 15px 18px;
+    background: var(--90_white);
+    border-radius: 10px;
+    border: none;
+    align-items: center;
+    text-align: center;
+    color: var(--jet-black);
+    display: grid;
+    grid-template-rows: 1fr 40px;
+    grid-gap: 5px;
```
Лучше располагать стили по их важности - text-decoration в конец, а дисплеи всякие повыше

DPeshkoff commented on index.css:
```diff
@@ -0,0 +1,211 @@
+
+.item-card__rating {
+    background: transparent;
+    border: none !important;
```
опять какие-то импортанты :с

DPeshkoff reviewed (changes requested):
Привет! Есть что поправить + следите, чтобы соответствовало макету

DPeshkoff commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Base-1.3: Сверстать home page](https://trello.com/c/lVOd4UHN/5-base-13-%D1%81%D0%B2%D0%B5%D1%80%D1%81%D1%82%D0%B0%D1%82%D1%8C-home-page)

tUnknownLegend committed:
убраны пустые дивы, исправлена строка поиска

по возможности выбраны кратные размеры, выравнены и уравнены в размере правые кнопки в хедере

tUnknownLegend committed:
merge from Base-1.2_html_make_up

tUnknownLegend committed:
Добавлен скрин верстки

DPeshkoff commented on index.css:
```diff
@@ -0,0 +1,196 @@
+}
+
+.arrow-button-right {
+    margin-top: 171px;
```
171?

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,523 @@
+            </div>
+            <div class="header__catalog">
+                <a href="" class="button-default header__button-catalog paint-purple">
+                    <img type="image" src="./img/Catalog.svg" class="header__catalog-img" alt="CAT">
```
alt

DPeshkoff commented on index.html:
```diff
@@ -0,0 +1,523 @@
+
+                <div type="left-arrow">
+                    <button type="button" class="arrow-button arrow-button-left text-normal-default">
+                        <img src="img/Left-arrow.svg" alt="<-">
```
лучше текстом - альт-текст читают скринридеры
